### ISSUE TITLE
changed parameters don't stick to models - bug or feature?

### ISSUE THREAD

**Issue by nden** [astropy/astropy#4819]
This came up in `gwcs`. If I change a parameter of a compound model and then combine it with another model, the resulting model has the orignal parameter value.

```
scl=models.Scale(1) & models.Scale(2)
>>> rot = models.Rotation2D(23)
>>> scl
<CompoundModel0(factor_0=1.0, factor_1=2.0)>
>>> (rot | scl)(1,1)
(0.5297737249631667, 2.622471963883428)
>>> scl.factor_1 = 10
>>> scl
<CompoundModel0(factor_0=1.0, factor_1=10.0)>
>>> (rot | scl)(1,1)
(0.5297737249631667, 2.622471963883428)
>>> (rot | scl)
<CompoundModel3(angle_0=23.0, factor_1=1.0, factor_2=2.0)>
>>> scl
<CompoundModel0(factor_0=1.0, factor_1=10.0)>
>>> scl._get_submodels()
[<Scale(factor=1.0)>, <Scale(factor=2.0)>]
>>> scl.parameters
array([  1.,  10.])
```

I'd say it's a bug but it looks like it was designed this way.

**Comment by nden** [comment]
@embray if available

**Comment by pllim** [comment]
This will be fixed by #5741, right?

**Comment by nden** [comment]
@pllim  yes, #5741 is a replacement for this.
